File: src/integration/spring/src/jvmTest/java/community/flock/wirespec/integration/spring/java/it/client/WirespecPetstoreWebClient.java
package community.flock.wirespec.integration.spring.java.it.client;

import community.flock.wirespec.integration.spring.java.client.WirespecWebClient;
import community.flock.wirespec.integration.spring.java.generated.endpoint.AddPet;
import community.flock.wirespec.integration.spring.java.generated.endpoint.DeletePet;
import community.flock.wirespec.integration.spring.java.generated.endpoint.FindPetsByTags;
import community.flock.wirespec.integration.spring.java.generated.endpoint.GetPetById;
import community.flock.wirespec.integration.spring.java.generated.endpoint.UpdatePet;
import community.flock.wirespec.integration.spring.java.generated.endpoint.UploadFile;
import org.springframework.stereotype.Component;

import java.util.concurrent.CompletableFuture;

@Component
public class WirespecPetstoreWebClient implements
        AddPet.Handler,
        UpdatePet.Handler,
        GetPetById.Handler,
        DeletePet.Handler,
        FindPetsByTags.Handler,
        UploadFile.Handler {

    private final WirespecWebClient wirespecWebClient;

    public WirespecPetstoreWebClient(WirespecWebClient wirespecWebClient) {
        this.wirespecWebClient = wirespecWebClient;
    }

    @Override
    public CompletableFuture<GetPetById.Response<?>> getPetById(GetPetById.Request request) {
        return wirespecWebClient.send(request);
    }

    @Override
    public CompletableFuture<AddPet.Response<?>> addPet(AddPet.Request request) {
        return wirespecWebClient.send(request);
    }

    @Override
    public CompletableFuture<UpdatePet.Response<?>> updatePet(UpdatePet.Request request) {
        return wirespecWebClient.send(request);
    }

    @Override
    public CompletableFuture<DeletePet.Response<?>> deletePet(DeletePet.Request request) {
        return wirespecWebClient.send(request);
    }

    @Override
    public CompletableFuture<FindPetsByTags.Response<?>> findPetsByTags(FindPetsByTags.Request request) {
        return wirespecWebClient.send(request);
    }

    @Override
    public CompletableFuture<UploadFile.Response<?>> uploadFile(UploadFile.Request request) {
        return wirespecWebClient.send(request);
    }
}

File: src/integration/spring/src/jvmTest/kotlin/community/flock/wirespec/integration/spring/kotlin/it/client/WirespecPetstoreWebClient.kt
package community.flock.wirespec.integration.spring.kotlin.it.client

import community.flock.wirespec.integration.spring.kotlin.client.WirespecWebClient
import community.flock.wirespec.integration.spring.kotlin.generated.endpoint.AddPet
import community.flock.wirespec.integration.spring.kotlin.generated.endpoint.DeletePet
import community.flock.wirespec.integration.spring.kotlin.generated.endpoint.FindPetsByTags
import community.flock.wirespec.integration.spring.kotlin.generated.endpoint.GetPetById
import community.flock.wirespec.integration.spring.kotlin.generated.endpoint.UpdatePet
import community.flock.wirespec.integration.spring.kotlin.generated.endpoint.UploadFile
import org.springframework.stereotype.Component

@Component
class WirespecPetstoreWebClient(
    private val wirespecWebClient: WirespecWebClient,
) : AddPet.Handler,
    UpdatePet.Handler,
    GetPetById.Handler,
    DeletePet.Handler,
    FindPetsByTags.Handler,
    UploadFile.Handler {
    override suspend fun getPetById(request: GetPetById.Request): GetPetById.Response<*> = wirespecWebClient.send(request)

    override suspend fun addPet(request: AddPet.Request): AddPet.Response<*> = wirespecWebClient.send(request)

    override suspend fun updatePet(request: UpdatePet.Request): UpdatePet.Response<*> = wirespecWebClient.send(request)

    override suspend fun deletePet(request: DeletePet.Request): DeletePet.Response<*> = wirespecWebClient.send(request)

    override suspend fun findPetsByTags(request: FindPetsByTags.Request): FindPetsByTags.Response<*> = wirespecWebClient.send(request)

    override suspend fun uploadFile(request: UploadFile.Request): UploadFile.Response<*> = wirespecWebClient.send(request)
}
